controller/volume_controller.go

```diff
@@ -2853,7 +2853,21 @@ func (vc *VolumeController) checkForAutoDetachment(v *longhorn.Volume, e *longho
 	}
 
 	if v.Status.ExpansionRequired {
-		return nil
+		if v.Status.CurrentNodeID != "" {
+			_, err := vc.ds.GetNodeRO(v.Status.CurrentNodeID)
+			if err == nil {
+				log.Debugf("Expansion of volume %v is required", v.Name)
+				return nil
+			}
+
+			if err != nil && !datastore.ErrorIsNotFound(err) {
+				log.WithError(err).Debugf("Failed to get node %v for checking volume %v detachment", v.Status.CurrentNodeID, v.Name)
+				return nil
+			}
+
+			v.Status.CurrentNodeID = ""
```

Comment:
Add a log before setting the `CurrentNodeID` to empty, something like `Preparing to do auto detachment of volume xx, because the node xx is unavailable`.

Reply:
Updated. Thank you.